geomesa-fs/geomesa-fs-spark/src/main/scala/org/locationtech/geomesa/fs/spark/ParquetFileSystemRDD.scala

```diff
@@ -0,0 +1,114 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.fs.spark
+
+import java.io.Serializable
+import java.util
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+import org.apache.parquet.hadoop.ParquetInputFormat
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.{DataStoreFinder, Query, Transaction}
+import org.locationtech.geomesa.fs.{FileSystemDataStore, FsQueryPlanning}
+import org.locationtech.geomesa.index.planning.QueryPlanner
+import org.locationtech.geomesa.parquet.{FilterConverter, ParquetFileSystemStorageFactory, SFParquetInputFormat, SimpleFeatureReadSupport}
+import org.locationtech.geomesa.spark.{SpatialRDD, SpatialRDDProvider}
+import org.locationtech.geomesa.utils.geotools.FeatureUtils
+import org.opengis.feature.simple.SimpleFeature
+
+class ParquetFileSystemRDD extends SpatialRDDProvider with LazyLogging {
```

Comment:
We need a partition-aware, Converter-powered RDDProvider.  That'd really help the GDELT stuff to sing.